### PR TITLE
fix(onnx): keep the encoder output sequence dimension dynamic

### PR DESCRIPTION
**Pull request by YageGeng**
Refs #12.

The encoder ONNX export declares four output positions even though a 384×384 input produces 144. This causes ONNX Runtime output-shape mismatch warnings for normal inference.

`HGNetv2OnnxConfig` inherits ViT's batch-only output axes. The composite `VisionEncoderDecoderOnnxConfig` uses dynamic spatial input axes but delegates its encoder outputs to that configuration. Optimum's dynamic-axis fix consequently freezes the output sequence length to the four positions produced by the default 64×64 dummy input.

This change explicitly marks `last_hidden_state` axis 1 as `encoder_sequence_length`, keeping the output metadata consistent with the existing dynamic spatial inputs. It does not hard-code 144 or change the model's computation or weights.

The independent regression test:

- Exports the real HGNetv2 architecture with seeded random weights through the composite encoder ONNX configuration, including the default dynamic-axis fix; no checkpoint download is needed.
- Checks the original ONNX output metadata and runs `onnx.checker`.
- Runs CPU inference at 384×384 for batch sizes 1 and 4, checking both PyTorch and ONNX Runtime output shapes and the absence of output-shape mismatch warnings.

Validation:

```bash
PYTHONPATH=src python -m pytest -q tests/test_encoder_onnx_shape.py
```

Before the fix: both cases fail with `4 != 144` and reproduce the reported warning. After the fix: both pass; metadata is `[batch_size, encoder_sequence_length, 2048]`, and runtime outputs are `[1, 144, 2048]` and `[4, 144, 2048]`.

Tested on Linux / Python 3.12 with PyTorch 2.7.0+cpu, Transformers 4.40.0, Optimum 2.0.0, optimum-onnx 0.0.1, ONNX 1.22.0 and ONNX Runtime 1.30.0 (CPU). Existing unrelated exporter warnings remain. Full pretrained trio export, OCR accuracy and GPU/WebGPU inference have not been tested in this change.

### Required follow-up by the Texo repository owner

After merging this PR, the Texo repository owner needs to re-export the ONNX models from the pretrained checkpoint using the updated exporter, validate the resulting encoder with batch sizes 1 and 4 at 384×384, and publish the regenerated artifacts to the [alephpi/FormulaNet model repository on Hugging Face](https://huggingface.co/alephpi/FormulaNet), including the corrected `onnx/encoder_model.onnx`.

Merging this source-code change alone does not update the model files already hosted on Hugging Face. Users downloading the existing artifact will continue to see the warning until the updated models are published; users pinned to an older revision or using a cached local copy will also need to download the new revision.

This PR changes the exporter and adds standalone regression coverage; it does not add inference checks to the normal export command.

### Additional batch-size validation

Additional manual checks on the same fixed encoder export passed with 384×384 inputs using ONNX Runtime's CPUExecutionProvider:

| Batch size | Actual `last_hidden_state` shape | Result |
| --- | --- | --- |
| 2 | `[2, 144, 2048]` | Passed |
| 3 | `[3, 144, 2048]` | Passed |
| 8 | `[8, 144, 2048]` | Passed |

All three runs completed without output-shape mismatch warnings. Together with the automated batch=1 and batch=4 cases, this verifies five batch sizes. These additional checks used the same randomly initialized HGNetv2 encoder export; they do not validate pretrained OCR accuracy or full decoder/generate batching. Larger batches remain subject to memory limits. The committed regression test remains parameterized for batch sizes 1 and 4.

### CODE REVIEW

**Comment by alephpi** [comment]
thx!
